src/components/todolist/todolist.tsx

```diff
@@ -0,0 +1,292 @@
+      <div className="toast">
+        <Toast appearance={toastType} title={title} />
+      </div>
+    );
+  };
+
+  const captureForm = () => {
+    return (
+      <Formik
+        initialValues={{
+          todo: selectedItem.oldValue ? selectedItem.oldValue : "",
+        }}
+        enableReinitialize={true}
+        onSubmit={(values, { resetForm }) => {
+          if (selectedItem.id && props.editTodoDispatch) {
+            props.editTodoDispatch({ id: selectedItem.id, item: values.todo });
+            setSelectedItem({ oldValue: "", updatedValue: "", id: "" });
+          } else {
+            addTodo(values.todo);
+          }
+          resetForm();
+        }}
+      >
+        {(props) => {
+          const {
+            values,
+            touched,
+            errors,
+            dirty,
+            isSubmitting,
+            handleChange,
+            handleBlur,
+            handleSubmit,
+            handleReset,
+          } = props;
+          return (
+            <form onSubmit={handleSubmit}>
+              <label
+                htmlFor="todo"
+                style={{
+                  textAlign: "left",
+                  display: "block",
+                  fontWeight: "bold",
+                  marginBottom: 10,
+                }}
+              >
+                Add a task
+              </label>
+              <div>
+                <input
+                  id="todo"
+                  className="py-5 px-6"
+                  placeholder="New Task"
+                  type="text"
+                  value={values.todo}
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                />
+              </div>
+              <div className="d-flex flex-row justify-content-start mt-7">
+                <Button
+                  appearance="alert"
+                  type="button"
+                  className="outline mr-6"
+                  onClick={handleReset}
+                  disabled={!dirty || isSubmitting}
+                >
+                  Clear
+                </Button>
+                <Button appearance="primary" type="submit">
+                  {selectedItem.id ? "Update" : "Add"}
+                </Button>
+              </div>
+            </form>
+          );
+        }}
+      </Formik>
+    );
+  };
+
+  return (
+    <div className="mt-10">
+      <h1>Todo List</h1>
+      <div className="d-flex align-items-start flex-row mt-8 pb-4 ">
+        <div className="mx-10">{captureForm()}</div>
+        <div
+          style={{
+            display: "flex",
+            height: "400px",
+            borderWidth: "1px",
+            borderStyle: "solid",
+            borderColor: "gray",
+          }}
+        ></div>
+
+        <div
+          style={{
+            width: "30%",
+            overflowWrap: "break-word",
+            marginLeft: 70,
+          }}
+        >
+          <DisplayTodos />
+        </div>
+        {actionPerformed && showToast()}
```

Comment:
```suggestion
        {showToast && renderToast()}
```

OR 

Create a component that uses your implementation of Toast
```suggestion
        {showToast && <Toast />}
```